<a href="https://colab.research.google.com/github/oimartin/Older-and-Wiser/blob/geriatric_visits/geriatric_visits_nb.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Injest

In [1]:
import pandas as pd
import requests

In [2]:
visits = pd.read_csv("https://raw.githubusercontent.com/oimartin/Older-and-Wiser/geriatric_visits/data/Geriatric_Medicine.csv")
zip_codes = pd.read_csv('https://raw.githubusercontent.com/oimartin/Older-and-Wiser/geriatric_visits/data/all_states_only_zip_codes.csv')

# EDA

## Geriatric Visits

In [3]:
visits.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 42959 entries, 0 to 42958
Data columns (total 15 columns):
 #   Column                                                Non-Null Count  Dtype  
---  ------                                                --------------  -----  
 0   zip_code                                              42959 non-null  int64  
 1   min_medicare_pricing_for_new_patient                  42959 non-null  float64
 2   max_medicare_pricing_for_new_patient                  42959 non-null  float64
 3   mode_medicare_pricing_for_new_patient                 42959 non-null  float64
 4   min_copay_for_new_patient                             42959 non-null  float64
 5   max_copay_for_new_patient                             42959 non-null  float64
 6   mode_copay_for_new_patient                            42959 non-null  float64
 7   most_utilized_procedure_code_for_new_patient          42959 non-null  int64  
 8   min_medicare_pricing_for_established_patient          42

In [11]:
visits.iloc[:, 0] = visits['zip_code'].astype('string')
visits.iloc[:, 7] = visits.iloc[:, 7].astype('string')
visits.iloc[:, 14] = visits.iloc[:, 14].astype('string')

In [4]:
visits.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 42959 entries, 0 to 42958
Data columns (total 15 columns):
 #   Column                                                Non-Null Count  Dtype  
---  ------                                                --------------  -----  
 0   zip_code                                              42959 non-null  int64  
 1   min_medicare_pricing_for_new_patient                  42959 non-null  float64
 2   max_medicare_pricing_for_new_patient                  42959 non-null  float64
 3   mode_medicare_pricing_for_new_patient                 42959 non-null  float64
 4   min_copay_for_new_patient                             42959 non-null  float64
 5   max_copay_for_new_patient                             42959 non-null  float64
 6   mode_copay_for_new_patient                            42959 non-null  float64
 7   most_utilized_procedure_code_for_new_patient          42959 non-null  int64  
 8   min_medicare_pricing_for_established_patient          42

In [5]:
distrib_df = visits.describe()
distrib_df

zip_code  min_medicare_pricing_for_new_patient  \
count  42959.000000                          42959.000000   
mean   49591.379245                             58.238339   
std    27974.499582                              4.045550   
min      210.000000                             53.136000   
25%    26037.500000                             55.312000   
50%    48815.000000                             56.936000   
75%    73042.500000                             59.432000   
max    99950.000000                             74.816000   

       max_medicare_pricing_for_new_patient  \
count                          42959.000000   
mean                             177.239252   
std                               11.124559   
min                              163.672000   
25%                              169.736000   
50%                              174.056000   
75%                              180.408000   
max                              233.632000   

       mode_medicare_pricing_for_new_patient  min_copay_for_new_patient  \
count                           42959.000000               42959.000000   
mean                              177.239252                  14.559585   
std                                11.124559                   1.011387   
min                               163.672000                  13.284000   
25%                               169.736000                  13.828000   
50%                               174.056000                  14.234000   
75%                               180.408000                  14.858000   
max                               233.632000                  18.704000   

       max_copay_for_new_patient  mode_copay_for_new_patient  \
count               42959.000000                42959.000000   
mean                   44.309813                   44.309813   
std                     2.781140                    2.781140   
min                    40.918000                   40.918000   
25%                    42.434000                   42.434000   
50%                    43.514000                   43.514000   
75%                    45.102000                   45.102000   
max                    58.408000                   58.408000   

       most_utilized_procedure_code_for_new_patient  \
count                                       42959.0   
mean                                        99205.0   
std                                             0.0   
min                                         99205.0   
25%                                         99205.0   
50%                                         99205.0   
75%                                         99205.0   
max                                         99205.0   

       min_medicare_pricing_for_established_patient  \
count                                  42959.000000   
mean                                      18.073701   
std                                        1.459078   
min                                       16.256000   
25%                                       17.040000   
50%                                       17.616000   
75%                                       18.544000   
max                                       23.768000   

       max_medicare_pricing_for_established_patient  \
count                                  42959.000000   
mean                                     144.793623   
std                                        9.072558   
min                                      133.848000   
25%                                      138.800000   
50%                                      142.080000   
75%                                      147.312000   
max                                      190.744000   

       mode_medicare_pricing_for_established_patient  \
count                                   42959.000000   
mean                                      103.643492   
std                                         6.628021   
min                                        95.608000   
25%             

In [14]:
print("most_utilized_procedure_code_for_new_patient")
display(visits.iloc[:, 7].value_counts())
print("most_utilized_procedure_code_for_established_patient")
display(visits.iloc[:, 14].value_counts())

most_utilized_procedure_code_for_new_patient


99205    42959
Name: most_utilized_procedure_code_for_new_patient, dtype: Int64

most_utilized_procedure_code_for_established_patient


99214    42959
Name: most_utilized_procedure_code_for_established_patient, dtype: Int64

In [15]:
visits.nunique()

zip_code                                                42959
min_medicare_pricing_for_new_patient                       95
max_medicare_pricing_for_new_patient                       96
mode_medicare_pricing_for_new_patient                      96
min_copay_for_new_patient                                  95
max_copay_for_new_patient                                  96
mode_copay_for_new_patient                                 96
most_utilized_procedure_code_for_new_patient                1
min_medicare_pricing_for_established_patient               90
max_medicare_pricing_for_established_patient               96
mode_medicare_pricing_for_established_patient              96
min_copay_for_established_patient                          90
max_copay_for_established_patient                          96
mode_copay_for_established_patient                         96
most_utilized_procedure_code_for_established_patient        1
dtype: int64

99214: Established patient office visit, 30-39 minutes
<br>
99205: Typically 60 minutes, Comprehensive history, Comprehensive Examination, High complexity medical decision making

## Zip Codes

In [4]:
zip_codes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 27695 entries, 0 to 27694
Data columns (total 3 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   STATEFIPS  27695 non-null  int64 
 1   STATE      27695 non-null  object
 2   ZIPCODE    27695 non-null  int64 
dtypes: int64(2), object(1)
memory usage: 649.2+ KB


In [5]:
zip_codes.nunique()

STATEFIPS       51
STATE           51
ZIPCODE      27595
dtype: int64

In [8]:
zip_codes.head()

STATEFIPS STATE  ZIPCODE
0          1    AL        0
1          1    AL    35004
2          1    AL    35005
3          1    AL    35006
4          1    AL    35007

In [8]:
clean_zip = zip_codes[(zip_codes['ZIPCODE'] !=0) &
                      (zip_codes['ZIPCODE'] !=99999)]
clean_zip.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 27593 entries, 1 to 27693
Data columns (total 3 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   STATEFIPS  27593 non-null  int64 
 1   STATE      27593 non-null  object
 2   ZIPCODE    27593 non-null  int64 
dtypes: int64(2), object(1)
memory usage: 862.3+ KB


In [9]:
clean_zip.nunique()

STATEFIPS       51
STATE           51
ZIPCODE      27593
dtype: int64

In [26]:
clean_zip.iloc[:, 2] = clean_zip['ZIPCODE'].astype('string')
clean_zip.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 27644 entries, 1 to 27694
Data columns (total 3 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   STATEFIPS  27644 non-null  int64 
 1   STATE      27644 non-null  object
 2   ZIPCODE    27644 non-null  string
dtypes: int64(1), object(1), string(1)
memory usage: 863.9+ KB


/usr/local/lib/python3.7/dist-packages/pandas/core/indexing.py:1773: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._setitem_single_column(ilocs[0], value, pi)


In [10]:
clean_zip.describe()

STATEFIPS       ZIPCODE
count  27593.000000  27593.000000
mean      29.669663  48856.241474
std       15.119769  27047.660765
min        1.000000   1001.000000
25%       18.000000  27105.000000
50%       29.000000  48845.000000
75%       42.000000  70578.000000
max       56.000000  99901.000000

In [21]:
def find_start_end_zip(state):
  zipcodes_state = clean_zip[clean_zip['STATE'] == state]['ZIPCODE']
  min_zip = zipcodes_state.min()
  max_zip = zipcodes_state.max()

  return min_zip, max_zip

In [22]:
ME_min, ME_max = find_start_end_zip('ME')
display(ME_min, ME_max)

3901

4989

In [24]:
clean_zip.rename(columns={'ZIPCODE': 'zip_code'},inplace=True)

/usr/local/lib/python3.7/dist-packages/pandas/core/frame.py:5047: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  errors=errors,


## Combine Visits and States/Zip Codes

In [25]:
visits.join(clean_zip, on='zip_code', how='left', )

ValueError: ignored

# Aggregate

In [16]:
visits.sort_values(by='min_copay_for_new_patient')

zip_code  min_medicare_pricing_for_new_patient  \
32170    72950                                53.136   
31710    72117                                53.136   
31711    72118                                53.136   
31712    72119                                53.136   
31713    72120                                53.136   
...        ...                                   ...   
42771    99622                                74.816   
42770    99621                                74.816   
42769    99620                                74.816   
42775    99626                                74.816   
42958    99950                                74.816   

       max_medicare_pricing_for_new_patient  \
32170                               163.672   
31710                               163.672   
31711                               163.672   
31712                               163.672   
31713                               163.672   
...                                     ...   
42771                               233.632   
42770                               233.632   
42769                               233.632   
42775                               233.632   
42958                               233.632   

       mode_medicare_pricing_for_new_patient  min_copay_for_new_patient  \
32170                                163.672                     13.284   
31710                                163.672                     13.284   
31711                                163.672                     13.284   
31712                                163.672                     13.284   
31713                                163.672                     13.284   
...                                      ...                        ...   
42771                                233.632                     18.704   
42770                                233.632                     18.704   
42769                                233.632                     18.704   
42775                                233.632                     18.704   
42958                                233.632                     18.704   

       max_copay_for_new_patient  mode_copay_for_new_patient  \
32170                     40.918                      40.918   
31710                     40.918                      40.918   
31711                     40.918                      40.918   
31712                     40.918                      40.918   
31713                     40.918                      40.918   
...                          ...                         ...   
42771                     58.408                      58.408   
42770                     58.408                      58.408   
42769                     58.408                      58.408   
42775                     58.408                      58.408   
42958                     58.408                      58.408   

      most_utilized_procedure_code_for_new_patient  \
32170                                        99205   
31710                                        99205   
31711                                        99205   
31712                                        99205   
31713                                        99205   
...                                            ...   
42771                                        99205   
42770                                        99205   
42769                                        99205   
42775                                        99205   
42958                                        99205   

       min_medicare_pricing_for_established_patient  \
32170                                        16.264   
31710                                        16.264   
31711                                        16.264   
31712                                        16.264   
31713                                        16.264   
...                                             ...   
42771                                        22.376   
42770    